tests/CacheCompat/CommonCache.Test.MsalV3/FileBasedTokenCacheHelper.cs

```diff
@@ -0,0 +1,124 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System.IO;
+using System.Security.Cryptography;
+using CommonCache.Test.Common;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Cache;
+
+namespace CommonCache.Test.MsalV2
+{
+    /// <summary>
+    ///     Simple persistent cache implementation of the dual cache serialization (ADAL V3 legacy
+    ///     and unified cache format) for a desktop applications (from MSAL 2.x)
+    /// </summary>
+    public static class FileBasedTokenCacheHelper
+    {
+        private static readonly object s_fileLock = new object();
+        private static CacheStorageType s_cacheStorageType = CacheStorageType.None;
+
+        public static string MsalV2CacheFileName { get; private set; }
+        public static string MsalV3CacheFileName { get; private set; }
+        public static string AdalV3CacheFileName { get; private set; }
+
+        public static void ConfigureUserCache(
```

Comment:
Wouldn't there be value in exposing FileBasedTokenCacheHelper in the library (as a helper) to customers (if it crypted data, of course)

Reply:
We could look at providing this as a helper class.  Do we need to do API review and inclusion for library consistency or is this just something we can do to help our customers?  (hoping for the latter)